### PR TITLE
application check in and out

### PR DESCRIPTION
**Pull request by DiegoCoronaBarragan**
# Description

We implemented the check-in/out of the app and edited some tables for their proper functioning.

## Screenshots

image

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -119,4 +121,10 @@ input[type="submit" i] {
...
   margin-top: 50px;
   text-align: center;
-  }   
+  }
+.check__buttons_container {
+  display: flex;
+  justify-content: space-between;
+  text-align: center;
+
+}
```
Comment (Reiyirak) [inline app/assets/stylesheets/application.css:130]: Add a line at the end of the file

File: app/controllers/front_view_controller.rb [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+  private
+
+  def employee_params
+    params.permit(:employee_num, :employee_private_num)
+  end
+
+  def check_params
+    params.permit(:check)
   end
 end
```
Comment (Reiyirak) [inline app/controllers/front_view_controller.rb:28]: Add an empty line at the end

File: app/models/assitance.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class Assitance < ApplicationRecord
+end
```
Comment (Reiyirak) [inline app/models/assitance.rb:2]: Add an empty line at the end

File: app/models/employee.rb [hunk 4]
```diff
@@ -1,2 +1,3 @@
 class Employee < ApplicationRecord
+    has_many :front_views, foreign_key:"employee_id"
 end
```
Comment (Reiyirak) [inline app/models/employee.rb:3]: Add an empty line at the end

File: app/models/front_view.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class FrontView < ApplicationRecord
+  belongs_to :employee, foreign_key:"employee_id"
+end
```
Comment (Reiyirak) [inline app/models/front_view.rb:3]: Add an empty line at the end

File: app/views/front_view/index.html.erb [hunk 6]
```diff
@@ -34,13 +34,35 @@
...
+          <% end %>
+        </div>
+
+        <div class="check__buttons_container pt-1">
+          <%= form.submit "Check in", name:"check", value:"in", class: "check__buttons_container__in " %>
+          <%= form.submit "Check off", name:"check", value:"off",  class: "check__buttons_container__off " %>
+        </div>
+      <% end %>
+  </div>
 </div>
```
Comment (Reiyirak) [inline app/views/front_view/index.html.erb:68]: Add an empty line at the end

File: app/views/employees/index.html.erb [hunk 7]
```diff
@@ -28,7 +28,7 @@
               <td><%= link_to 'Show', employee_path(employee.id) %></td>
               <td><%= link_to 'Edit', edit_employee_path(employee.id) %></td>
               <td><%= button_to 'Delete', employee_path(employee.id), method: :delete %></td>
-              
+
             </tr>
           <% end %>
         </tbody>
```
Comment (Reiyirak) [inline app/views/employees/index.html.erb:34]: Add an empty line at the end

File: config/routes.rb [hunk 8]
```diff
@@ -1,18 +1,19 @@
...
-
 
   get 'admin/admin_menu'
 
   resources :employees
 
   resources :stores
 
-  root 'front_view#index'
+  resources :front_view
```
Comment (Reiyirak) [inline config/routes.rb:18]: Add an empty line at the end

File: db/migrate/20230127231203_create_front_views.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+class CreateFrontViews < ActiveRecord::Migration[7.0]
+  def change
+    create_table :front_views do |t|
+      t.string :check
+      t.belongs_to :employee, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```
Comment (Reiyirak) [inline db/migrate/20230127231203_create_front_views.rb:10]: Add an empty line at the end

File: db/migrate/20230130215427_change_data_type.rb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+class ChangeDataType < ActiveRecord::Migration[7.0]
+  def change
+    remove_column(:employees, :employee_num)
+    remove_column(:employees, :employee_private_num)
+  end
+end
```
Comment (Reiyirak) [inline db/migrate/20230130215427_change_data_type.rb:6]: Add an empty line at the end

File: db/migrate/20230130221325_add_columns.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+class AddColumns < ActiveRecord::Migration[7.0]
+  def change
+    add_column :employees, :employee_num, :integer
+    add_column :employees, :employee_private_num, :integer
+  end
+end
```
Comment (Reiyirak) [inline db/migrate/20230130221325_add_columns.rb:6]: Add an empty line at the end

File: db/migrate/20230130235713_create_assitances.rb [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+      t.integer :assit
+      t.integer :absence
+      t.time :checkin
+      t.time :checkout
+      t.belongs_to :front_views, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```
Comment (Reiyirak) [inline db/migrate/20230130235713_create_assitances.rb:13]: Add an empty line at the end

File: test/models/front_view_test.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class FrontViewTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (Reiyirak) [inline test/models/front_view_test.rb:7]: Add an empty line at the end

**Review by Reiyirak — CHANGES_REQUESTED** [review]
Just the empty lines at the end of the files for the standard way. And it seems that the branch has some problems to be resolved before It can be merged.

File: app/views/front_view/index.html.erb [hunk 14]
```diff
@@ -18,13 +34,36 @@
...
+          <%= form.submit "Check in", name:"check", value:"in", class: "check__buttons_container__in " %>
+          <%= form.submit "Check off", name:"check", value:"off",  class: "check__buttons_container__off " %>
+        </div>
+      <% end %>
+  </div>
+</div>
 
-    <button>Check-in</button>
-  </form>
-</div>
```
Comment (Reiyirak) [inline app/views/front_view/index.html.erb:30]: Sorry maybe I didn't see it, but this one also needs an empty line at the end.

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -31,7 +30,6 @@
       </nav> 
       </header>
     <%end%>
-
     <%= yield %>
   </body>
 </html>
```
Comment (Reiyirak) [inline app/views/layouts/application.html.erb:35]: And also this one a line at the end.

**Review by Reiyirak — APPROVED** [review]
Just the two files left where you need to add an empty line. By the way, I think you need to add the development.log file to the .gitignore file. But maybe I'm wrong.
Good luck.

File: app/views/front_view/index.html.erb (hunk 14 above)
Comment (AMCernas) [reply app/views/front_view/index.html.erb:30]: there´s an empy line at the end of that file, don't know why github doesn't show it.

**Review by juperameza — APPROVED** [review]
Lgtm

**Review by MaxRetana — APPROVED** [review]
LGTM

**Pull request merged**